### PR TITLE
feat(PPDSC-2472): improve components tracking part 1

### PR DESCRIPTION
**Pull request by mutebg**
PPDSC-2472

**What**

1. Background - why this is needed
2. What did you do: added instrumentation to 
 - accordion
 - audio player
 - slider
 - select
 - selection-list
4. What does the reviewers should expect

**I have done:**
- [ ] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [ ] Updated relevant documentation

**I have tested manually:**
- [ ] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [ ] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

### CODE REVIEW

**Comment by newskit-engineering** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2472-improve-components-tracking-part-1
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2472-improve-components-tracking-part-1/storybook

File: src/audio-player-composable/__tests__/audio-player-composable.test.tsx [hunk 1]
```diff
@@ -1433,7 +1415,10 @@ describe('Audio Player Composable', () => {
       );
 
       const expectedObject = {
-        ...recordedTrackingOutputObject,
+        context: {
+          ...recordedTrackingOutputObject.context,
+          media_volume: 0,
```
Comment (baburay23) [inline src/audio-player-composable/__tests__/audio-player-composable.test.tsx:1420]: hey sorry ,I could not  figure out why you added the `media_volume `in for the
`should raise event while the audio is being played` test?

Comment (mutebg) [reply src/audio-player-composable/__tests__/audio-player-composable.test.tsx:1420]: media_volume and media_playback_rate are sent with every event, the same as duration, offset and other related data. I just extended the object

**Review by baburay23 — DISMISSED** [review]

File: src/test/test-utils.tsx [hunk 2]
```diff
@@ -21,11 +21,15 @@ export const renderWithImplementation: <T extends {}>(
   props?: T,
   fireEvent?: (event: InstrumentationEvent) => void,
   options?: Omit<RenderOptions, 'wrapper'>,
-) => RenderResult = (Component, props, fireEvent = () => {}, options) =>
+  theme?: UncompiledTheme | Theme,
```
Comment (Xin00163) [inline src/test/test-utils.tsx:24]: Is there a reason to add the theme here? Also we might need to update the test for this util fun

File: src/instrumentation/types.ts [hunk 3]
```diff
@@ -9,6 +9,7 @@ export enum EventTrigger {
   End = 'end',
   Pulse = 'pulse',
```
Comment (Xin00163) [inline src/instrumentation/types.ts:10]:
```suggestion
  Pause = 'pause',
```

**Review by Xin00163 — COMMENTED** [review]
Seems to be infinite loop of `{originator: 'audio-player-audio', trigger: 'pulse', context: {…}}` when visit audio-player on storybook 😬

File: src/instrumentation/types.ts (hunk 3 above)
Comment (mutebg) [reply src/instrumentation/types.ts:10]: this is not a pause, this is really the "pulse" like checking every second, its like that in the old player

Comment (Xin00163) [reply src/instrumentation/types.ts:10]: Ah sorry my bad

File: src/test/test-utils.tsx (hunk 2 above)
Comment (mutebg) [reply src/test/test-utils.tsx:24]: This is needed for slider tests, where a theme is needed. I don't see any coverage problems

Comment (Xin00163) [reply src/test/test-utils.tsx:24]: Actually you are right. I saw we were hardcoded it before.

**Comment by mutebg** [comment]
> Seems to be infinite loop of `{originator: 'audio-player-audio', trigger: 'pulse', context: {…}}` when visit audio-player on storybook 😬

the pulse fire event is sending on every second that's why is so noisy

**Review by Xin00163 — DISMISSED** [review]

**Review by baburay23 — APPROVED** [review]

**Review by Xin00163 — APPROVED** [review]

**Pull request merged**